ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/services/admin/ApiPackageManagementService.scala

```diff
@@ -76,13 +80,51 @@ class ApiPackageManagementService(
             case r @ UploadPackagesResult.NotSupported =>
               Future.failed(ErrorFactories.unimplemented(r.description))
           }(DE)
+          .flatMap(
+            pollUntilPersisted(
+              res._2,
+              _,
+              50.milliseconds,
+              500.milliseconds,
+              (d: FiniteDuration) => d * 2))(DE)
     )
   }
+
+  /**
+    * Wraps a call [[PollingUtils.pollUntilPersisted]] so that it can be chained on the package upload with a `flatMap`.
+    *
+    * Checks invariants and forwards the original result after all packages are found to be persisted.
+    *
+    * @param ids The IDs of the uploaded packages
+    * @return The result of the party allocation received originally, wrapped in a [[Future]]
+    */
+  private def pollUntilPersisted(
```

Comment:
Do you think it would make sense to move this in `PollingUtils`?

Reply:
If you prefer to leave it here, it may be worth to cut on the parameters and call `PollingUtils.pollUntilPersisted` with the actual values here.

Reply:
Since we use the same values for `minWait`, `maxWait`, and `backoffProgression`, we could also hard-code them inside `PollingUtils.pollUntilPersisted`. If we realize that we need different values for party and package checking, we can put them as parameters, but I doubt that will be the case.

Reply:
The test uses explicit values for `minWait`, `maxWait`, and `backoffProgression` to check whether they behave correctly, so I will leave the parameters for now.